concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -102,7 +102,11 @@ instance
                         case Vec.last versions of
                             EVersionedConfiguration vc -> vcGenesisHeight vc
                 let height = localToAbsoluteBlockHeight latestEraGenesisHeight (bpHeight bp)
-                liftIO (notifyCallback (bpHash bp) height)
+                ownBaker <- lift (asks (fmap (bakerId . bakerIdentity) . mvBaker))
+                let isHomeBaked = case ownBaker of
+                        Nothing -> False
+                        Just ourBakerId -> Just ourBakerId == (blockBaker <$> blockFields (_bpBlock bp))
+                liftIO (notifyCallback (bpHash bp) height isHomeBaked)
```

Comment:
This is not a comment about this code directly, but about an alternative.

It would be good to state here clearly as well (and not just in the prometheus documentation) that home-baked has a specific meaning about baker-ids and the caveat about catchup.

It would also be good to mention an alternative for this, namely that we could change `handleBlock` to take `homeBaked` as an argument which would directly tell us whether this is produced during baking or not. This is easy for handleBlock, however it is not easy for handleFinalize, although it woudl be possible to keep track of this, but that would add complexity.

Can you please add this comment here explaining why the alternative was not chosen as well?

Reply:
I have added a comment for `handleBlock` with the details, please review this